### PR TITLE
fix(forms): add DisplayName=TalendForm

### PR DESCRIPTION
**Pull request by jmfrancois**
**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our
  [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)

Form has no displayName so in test we have <Unkown />.

**What is the new behavior?**

Form.displayName = 'TalendForm';

### CODE REVIEW

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://381.talend.surge.sh)

File: packages/forms/src/Form.js [hunk 1]
```diff
@@ -191,4 +191,6 @@ Form.defaultProps = {
 	buttonBlockClass: 'form-actions',
 };
 
+Form.displayName = 'TalendForm';
```
Comment (frassinier) [inline packages/forms/src/Form.js]: What's the problem with default value `Form`?
https://facebook.github.io/react/docs/react-component.html#displayname

Comment (jmfrancois) [inline packages/forms/src/Form.js]: that default value happens only in dev. in prod the function name is minified.
And in snapshot without displayName I have <unknown />

https://github.com/Talend/data-streams/commit/d2554159919335e29312bf7b2b5e765e7c15f117#diff-8ac575987465edbb6a6b7810963bc2c8R24

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://381.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://381.talend.surge.sh)

**Pull request merged**